### DISCUSSION TITLE
v2raya 如何设置日志等级

### DISCUSSION THREAD

**Discussion by tinygrox** [v2rayA/v2rayA#1496] [Q&A] [answered]
有没有人知道怎么设置日志记录等级，默认设置的 info 太占空间，经常一不小心就几个 G，关键还全是一些没啥用的这种`[W] [asm_arm64.s:1222] ObservatoryProducer: did not connect: context deadline exceeded`信息，上下文都没有，设置还不能关。
之前还有 issue 的时候依稀记得搜到有一个环境变量是能设置的，但是现在找不到了，希望有大佬能看到，救救我的小硬盘吧。

**Comment by irort** [comment, marked as answer]
如果你用的一般linux发行版，编辑 /etc/default/v2raya , 取消 V2RAYA_LOG_LEVEL 前的注释，更改日志等级，保存，然后重启v2raya服务

**Reply by tinygrox** [reply]
虽然我用的是 Docker，没有/etc/default/v2raya这个玩意，但是我根据你的信息添加了运行参数 `-e V2RAYA_LOG_LEVEL=error`，现在搞定了。十分感谢！

**Comment by newptcai** [comment]
To be more specific, when you create the docker container v2raya, you can use the following

```
docker run -d \
  --restart=always \
  --privileged \
  --network=host \
  --name v2raya \
  -e V2RAYA_LOG_FILE=/tmp/v2raya.log \
  -e V2RAYA_V2RAY_BIN=/usr/local/bin/v2ray \
  -e V2RAYA_NFTABLES_SUPPORT=off \
  -e IPTABLES_MODE=legacy \
  -e V2RAYA_LOG_LEVEL=error \
  -v /lib/modules:/lib/modules:ro \
  -v /etc/resolv.conf:/etc/resolv.conf \
  -v /etc/v2raya:/etc/v2raya \
  mzz2017/v2raya
```

This will set the environment variable `V2RAYA_LOG_LEVEL` to `error`.

You will need to stop and remove the existing container before you can run it again with the updated environment variable. Here’s how to do that:

1. **Stop the Container**:
   ```bash
   docker stop v2raya
   ```

2. **Remove the Container**:
   ```bash
   docker rm v2raya
   ```
